hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieSparkUtils.scala

```diff
@@ -293,4 +293,21 @@ object HoodieSparkUtils extends SparkAdapterSupport {
       s"${tableSchema.fieldNames.mkString(",")}")
     AttributeReference(columnName, field.get.dataType, field.get.nullable)()
   }
+
+  def getRequiredSchema(tableAvroSchema: Schema, requiredColumns: Array[String]): (Schema, StructType) = {
+    // First get the required avro-schema, then convert the avro-schema to spark schema.
+    val name2Fields = tableAvroSchema.getFields.asScala.map(f => f.name() -> f).toMap
+    val requiredFields = requiredColumns.map(c => name2Fields(c))
+      .map(f => new Schema.Field(f.name(), f.schema(), f.doc(), f.defaultVal(), f.order())).toList
```

Comment:
just move this function from `MergeOnReadSnapshotRelation`. maybe it's not necessary.

Reply:
Here have to re-create the `Schema.Field`, can't use the origin object to `Schema.Field`. Otherwise, it will throw an exception like `org.apache.avro.AvroRuntimeException: Field already used: ts type:UNION pos:8`.

Reply:
Got it. Can you please add a comment elaborating that, so that it's very clear why it's this way